### PR TITLE
Add dynamic lobby tips.

### LINKED ISSUE #1724: [FEATURE] Dynamic tips on Lobby screen
## 🚀 Feature
The goal is to enhance the searching lobby screen and make it more dynamic. The changes include:

1. Displaying the selected character portrait.
2. Presenting a random tip.

For this initial iteration, let's utilize a scriptable object for managing the tips.

### PR DESCRIPTION
**Pull request by BertovDev**
Closes #1724 

Add dynamic tips and character Icon depending of the character you will use.

## Summary of changes

- Create tips.json with all the characters and general tips
- Read json and show tip according to the character you selected or a chance of 30% of showing a general tip.
- Also show the Icon of the character you selected.

## How has this been tested?

Describe the steps you performed to test this PR.

## Checklist
- [x] I have tested the changes locally.
- [x] I have tested the whole game after applying the changes, not only the affected areas.
- [x] I self-reviewed the changes on GitHub, line by line.
- [x] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [x] Tested in Android.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Review by alfopisano — DISMISSED** [review]
Nice!! Tested on unity desktop and iOS build and looks great

File: client/Assets/Resources/tips.json
Comment (AngieDutra) [inline client/Assets/Resources/tips.json:1]: Just looked at the code and saw this newline, it should be fixed
Screenshot 2024-05-09 at 19 40 36

**Review by AngieDutra — CHANGES_REQUESTED** [review]

**Comment by jazimp** [comment]
visual approval ✅

**Review by alfopisano — DISMISSED** [review]
Nice!! Tested on iOS build and everything seemed to work as expected

File: client/Assets/Scripts/GameNavigationManager.cs [hunk 1]
```diff
@@ -35,4 +42,64 @@ public void LeaveLobby()
...
+        foreach (CharacterTips character in charactersData.characters)
+        {
+            if(character.name.ToUpper() == characterSelected.ToUpper()){
+                int random = Random.Range(0,character.tips.Count());
+                string abilityTip = character.tips[random];
+
+                int threshold = Mathf.FloorToInt(character.tips.Count() * 0.3f);
+
+                if(random <= threshold){
+                    var genetalTip = charactersData.characters.Last();
```
Comment (AngieDutra) [inline client/Assets/Scripts/GameNavigationManager.cs:60]:
```suggestion
                    var generalTip = charactersData.characters.Last();
```

**Review by AngieDutra — CHANGES_REQUESTED** [review]

**Review by AngieDutra — APPROVED** [review]
Tested in Android build and is working as expected

File: client/Assets/Resources/tips.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        "name": "General",
+        "tips": [
+            "Low on health? Disengage from the battle and you'll see how you recover over time.",
+            "Don't chicken out: if you don't get power-ups it will be really hard to win the match.",
+            "Break them crates and get stronger and stronger!",
+            "Use the items well: they can make a difference between victory and defeat.",
+            "Don't let the sandstorm get you!",
+            "Use Magic Boots to run faster than the rest!",
+            "Use Myrra's Blessing and become immune to all damage!",
+            "Use Golden Clock to charge your ult and reset all cooldowns!"
```
Comment (mgrunwaldt) [inline client/Assets/Resources/tips.json:53]:
```suggestion
            "Use Golden Clock to charge your ulti and reset all cooldowns!"
```

**Review by mgrunwaldt — APPROVED** [review]

File: client/Assets/Resources/tips.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      {
+        "name": "General",
+        "tips": [
+            "Low on health? Disengage from the battle and you'll see how you recover over time.",
+            "Don't chicken out: if you don't get power-ups it will be really hard to win the match.",
+            "Break them crates and get stronger and stronger!",
+            "Use the items well: they can make a difference between victory and defeat.",
+            "Don't let the sandstorm get you!",
+            "Use Magic Boots to run faster than the rest!",
+            "Use Myrra's Blessing and become immune to all damage!",
```
Comment (AngieDutra) [inline client/Assets/Resources/tips.json:52]:
```suggestion
            "Use Mirra's Blessing and become immune to all damage!",
```

**Review by mgrunwaldt — APPROVED** [review]

**Review by AngieDutra — APPROVED** [review]
Awesome

**Pull request merged**